platforms/core-configuration/declarative-dsl-provider/src/integTest/groovy/org/gradle/internal/declarativedsl/settings/SoftwareTypeDeclarationIntegrationTest.groovy

```diff
@@ -0,0 +1,297 @@
+    }
+
+    /**
+     * This test is not yet implemented because it requires a custom repository to be set up which is not possible yet with the declarative dsl.
+     */
+    @NotYetImplemented
+    def 'can declare and configure a custom software type from plugin published to a custom repository'() {
+        given:
+        def pluginBuilder = withSoftwareTypePlugins()
+        pluginBuilder.publishAs("com", "example", "1.0", mavenHttpRepo, createExecuter()).allowAll()
+
+        file("settings.gradle.something") << """
+            pluginManagement {
+                repositories {
+                    maven { url("$mavenHttpRepo.uri") }
+                }
+            }
+            plugins {
+                id("com.example.test-software-type") version("1.0")
+            }
+        """
+
+        file("build.gradle.something") << declarativeScriptThatConfiguresSoftwareType
+
+        when:
+        run(":printConfiguration")
+
+        then:
+        assertThatDeclaredValuesAreSetProperly()
+    }
+
+    static String getDeclarativeScriptThatConfiguresSoftwareType() {
+        return """
+            testSoftwareType {
+                id = "test"
+
+                referencePoint = point(1, 2)
+
+                primaryAccess {
+                    read = false
+                    write = false
+                }
+
+                secondaryAccess {
+                    name = "two"
+                    read = true
+                    write = false
+                }
+
+                secondaryAccess {
+                    name = "three"
+                    read = true
+                    write = true
+                }
+            }
+        """
+    }
+
+    void assertThatDeclaredValuesAreSetProperly() {
+        outputContains("""id = test
+referencePoint = (1, 2)
+primaryAccess = { primary, false, false}
+secondaryAccess { two, true, false}
+secondaryAccess { three, true, true}"""
+        )
+    }
+
+    PluginBuilder withSoftwareTypePlugins() {
+        def pluginBuilder = new PluginBuilder(file("plugins"))
+        pluginBuilder.addPluginId("com.example.test-software-type-impl", "SoftwareTypeImplPlugin")
+        pluginBuilder.addPluginId("com.example.test-software-type", "SoftwareTypeRegistrationPlugin")
+
+        pluginBuilder.file("src/main/java/org/gradle/test/TestSoftwareTypeExtension.java") << """
+            package org.gradle.test;
+
+            import org.gradle.declarative.dsl.model.annotations.Adding;
+            import org.gradle.declarative.dsl.model.annotations.Configuring;
+            import org.gradle.declarative.dsl.model.annotations.Restricted;
+            import org.gradle.api.Action;
+            import org.gradle.api.model.ObjectFactory;
+            import org.gradle.api.provider.ListProperty;
+            import org.gradle.api.provider.Property;
+
+            import javax.inject.Inject;
+
+            @Restricted
+            public abstract class TestSoftwareTypeExtension {
```

Comment:
💭Minor suggestion, feel free to ignore in this PR: consider using a simpler extension that uses fewer Declarative DSL features. We test those features separately, and another test with all of them will make it harder to change things in the future.

Reply:
Good point.  I've simplified this to something much less complex.